### PR TITLE
fix Searching for a mxid in an encrypted room

### PR DESCRIPTION
**Pull request by EECvision**
### Bug fixes: 
** #Fixed Searching for a mxid in an encrypted room yields**

Fixes https://github.com/vector-im/element-desktop/issues/913

In the PR, I added a check to know when a user is searching for a mxid and to search for the id only. eg. 
when a user searches for `@userId:example.org`, the search term will be `@userId`

The code before:
before

The code after:
after

This links to the [issue](https://github.com/vector-im/element-desktop/issues/913) 

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * fix Searching for a mxid in an encrypted room ([\#8191](https://github.com/matrix-org/matrix-react-sdk/pull/8191)). Fixes vector-im/element-desktop#913. Contributed by @EECvision.

Preview: https://pr8191--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Comment by andybalaam** [comment]
@EECvision please add `Fixes https://github.com/vector-im/element-web/issues/21535` to the issue description so that github will know to close that issue if this PR is merged.

**Comment by EECvision** [comment]
This PR Fixes https://github.com/vector-im/element-web/issues/21535

File: src/components/structures/RoomView.tsx [hunk 1]
```diff
@@ -1319,6 +1319,13 @@ export class RoomView extends React.Component<IRoomProps, IRoomState> {
     }
 
     private onSearch = (term: string, scope: SearchScope) => {
+        // if we search for a user mxId or an organisation mxId, we need to
+        // search with only the userId or the organizationId.
+        const regex = /^@.*:|^#.*:/;
```
Comment (turt2live) [inline src/components/structures/RoomView.tsx:1324]: We should be trying to convert user IDs within the query to just their localparts, to fix issues where someone searches for `delete request @user:example.org` (for example) - the user ID will be in the middle of the query and still cause breakage. 

I believe we already have a regex somewhere which does this searching, it just needs to be found.

**Review by turt2live — CHANGES_REQUESTED** [review]

**Comment by andybalaam** [comment]
Abandoned - reopen if you are working on it.

**Pull request closed without merging**